lib/uplink/project.go

```diff
@@ -0,0 +1,204 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package uplink
+
+import (
+	"context"
+
+	"github.com/vivint/infectious"
+
+	"storj.io/storj/internal/memory"
+	"storj.io/storj/pkg/eestream"
+	"storj.io/storj/pkg/encryption"
+	"storj.io/storj/pkg/metainfo/kvmetainfo"
+	"storj.io/storj/pkg/storage/buckets"
+	ecclient "storj.io/storj/pkg/storage/ec"
+	"storj.io/storj/pkg/storage/segments"
+	"storj.io/storj/pkg/storage/streams"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/pkg/transport"
+	"storj.io/storj/uplink/metainfo"
+)
+
+// Project represents a specific project access session.
+type Project struct {
+	tc            transport.Client
+	metainfo      metainfo.Client
+	project       *kvmetainfo.Project
+	maxInlineSize memory.Size
+}
+
+// BucketConfig holds information about a bucket's configuration. This is
+// filled in by the caller for use with CreateBucket(), or filled in by the
+// library as Bucket.Config when a bucket is returned from OpenBucket().
+type BucketConfig struct {
+	// PathCipher indicates which ciphersuite is to be used for path
```

Comment:
"ciphersuite" is not a single word, right?

Reply:
It's often used as a single word these days, but yeah, I guess we capitalize it as though it's two words, so I should fix that